server/api/user/user.controller.js

```diff
@@ -577,3 +577,30 @@ exports.removeProject = function(req,res){
         }
     });
 };
+/*
+Function that is called by removeUser api call 
+
+*/
+exports.deleteUser = function(req,res,next){
+  var userId = req.params.id;
+  var pass = String(req.body.oldPassword);
+  var query = {students:{ $in: [userId]}};
+  console.log("hello");
+  User.findById(userId, function (err, user,db) {
+    if(user.authenticate(pass)) {
+       SmallGroup.findOneAndUpdate(query, {$pull: {address: userId}}, function(err, data){
```

Comment:
are you sure you are removing from the address field? aren't you removing from the list of students?